node/core/dispute-coordinator/src/real/mod.rs

```diff
@@ -499,10 +503,182 @@ async fn handle_new_activations(
 					db::v1::note_current_session(overlay_db, session)?;
 				}
 			},
-			_ => {},
+			Ok(SessionWindowUpdate::Unchanged) => {},
+		};
+		scrape_on_chain_votes(ctx, overlay_db, state, new_leaf, now, metrics).await?;
+	}
+
+	Ok(())
+}
+
+/// Scrapes on-chain votes (backing votes and concluded disputes) for a active leaf of the relay chain.
+async fn scrape_on_chain_votes(
+	ctx: &mut (impl SubsystemContext<Message = DisputeCoordinatorMessage>
+	          + overseer::SubsystemContext<Message = DisputeCoordinatorMessage>),
+	overlay_db: &mut OverlayedBackend<'_, impl Backend>,
+	state: &mut State,
+	new_leaf: Hash,
+	now: u64,
+	metrics: &Metrics,
+) -> Result<(), Error> {
+	// obtain the concluded disputes as well as the candidate backing votes
+	// from the new leaf
+	let ScrapedOnChainVotes { session, backing_validators, disputes } = {
+		let (tx, rx) = oneshot::channel();
+		ctx.send_message(RuntimeApiMessage::Request(
+			new_leaf,
+			RuntimeApiRequest::FetchOnChainVotes(tx),
+		))
+		.await;
+		match rx.await {
+			Ok(Ok(Some(val))) => val,
+			Ok(Ok(None)) => {
+				tracing::trace!(
+					target: LOG_TARGET,
+					relay_parent = ?new_leaf,
+					"No on chain votes stored for relay chain leaf");
+				return Ok(())
+			},
+			Ok(Err(e)) => {
+				tracing::trace!(
+					target: LOG_TARGET,
+					relay_parent = ?new_leaf,
+					error = ?e,
+					"Could not retrieve on chain votes due to an API error");
+				return Ok(())
+			},
+			Err(e) => {
+				tracing::trace!(
+					target: LOG_TARGET,
+					relay_parent = ?new_leaf,
+					error = ?e,
+					"Could not retrieve onchain votes due to oneshot cancellation");
+				return Ok(())
+			},
 		}
+	};
+
+	if backing_validators.is_empty() && disputes.is_empty() {
+		return Ok(())
 	}
 
+	// Obtain the session info, for sake of `ValidatorId`s
+	// either from the rolling session window or query from
+	// the on chain state directly
+	let session_info: SessionInfo =
+		if let Some(session_info) = state.rolling_session_window.session_info(session) {
+			session_info.clone()
+		} else {
+			let (tx, rx) = oneshot::channel();
+			ctx.send_message(RuntimeApiMessage::Request(
+				new_leaf,
+				RuntimeApiRequest::SessionInfo(session, tx),
+			))
+			.await;
+
+			match rx.await?? {
+				None => return Ok(()),
+				Some(session_info) => session_info,
+			}
+		};
+
+	// Scraped on-chain backing votes for the candidates with
+	// the new active leaf as if we received them via gossip.
+	for (candidate_receipt, backers) in backing_validators {
+		let candidate_hash = candidate_receipt.hash();
+		let statements = backers.into_iter().filter_map(|(validator_idx, validator_signature)| {
+			let validator_public: ValidatorId =
+				session_info.validators.get(validator_idx.0 as usize)?.clone();
+			let signed_dispute_statement = SignedDisputeStatement::new_checked(
```

Comment:
Sig verification here shouldn't be a bottleneck, but FWIW we can trust that the runtime has checked them already.

Reply:
There are two options here: Either spend the time to verify and maintaining our invariant of `SignedDisputeStatement`, or returning to by _convention_ and introducing a `fn from_on_chain_included`.

Reply:
Added `new_unchecked_from_trusted_source`